components/function-controller/pkg/apis/serverless/v1alpha2/function_types.go

```diff
@@ -118,6 +120,67 @@ type ResourceConfiguration struct {
 	Function *ResourceRequirements `json:"function,omitempty"`
 }
 
+type PodSpecTemplate struct {
+	// Compute Resources required by container created by function controller.
+	// +optional
+	Resources *ResourceRequirements `json:"resources,omitempty"`
```

Comment:
I believe on this level there should not be any "our" abstraction(i.e named presets "S", "L") but only k8s specs for  workload resources (v1.ResourceRequirements) that gets defaulted from the high level spec fields and/or can be fine tuned (customised ) here on this level

Reply:
We can add on root level field:
```
resourceProfiles:
  build: L
  runtime: S
```
and then default this template using profile.

Reply:
I dont think we need to add anything on root level but reuse those profiles that already exist , ie:
```yaml
spec:
  resourceConfiguration:
    function:
      profile: S
```

Reply:
I think that the structure is too big to what we want to achieve.
But we can leave it as it's if we don't want to change it.

Reply:
> I dont think we need to add anything on root level but reuse those profiles that already exist , ie:
> 
> ```yaml
> spec:
>   resourceConfiguration:
>     function:
>       profile: S
> ```

this structure will work only until the new API version is released. I think @dbadura is right we should add another structure that would represent the t-shirt functionality after removing the `resourceConfiguration` struct. It may look like:

User creates function:

```
...
spec:
  resourceConfiguration:
    function:
      profile: S
```

Webhook moves the `resourceConfiguration` to the new place and translates it to the right resources:

```
...
spec:
  resourceConfiguration:
    function:
      profile: S
  resourceProfiles:
    function: S
  templates:
    functionPodTemplate:
      spec:
        resources:
        [...]
```

The new API version (v1/v1beta1/v1alpha3) would look like this (after conversion):
```
spec:
  resourceProfiles:
    function: S
  templates:
    functionPodTemplate:
      spec:
        resources:
        [...]
```

Reply:
Here is another question. What do you think about NOT keeping old (high-level) fields after soft conversion? (soft means around one API spec version). 

Maybe we should accept high-level input from users and remove these fields after copying them into new ones. Then the example after moving and translating profiles by webhook would look like:

```
...
spec:
  resourceProfiles:
    function: S
  templates:
    functionPodTemplate:
      spec:
        resources:
        [...]
```

It looks exactly the same as the target spec we will introduce in the new version.